File: client/app/chat/[id]/agent-types.ts
import { WithMessages } from "@/hooks/useLangGraphAgent/types";

// The agent state which mirrors the LangGraph state. If your sate have messages, extend WithMessages interface.
export interface AgentState extends WithMessages {
  system_prompt?: string;
  weather_forecast: WeatherForecast[];
  context?: {
    agent_state: { flow_data?: string | null; service_id?: number | string | null };
    settings: Record<string, unknown>;
  };
}

export interface WeatherForecast {
  location: string;
  search_status: string;
  result: "Sunny" | "Cloudy" | "Rainy" | "Snowy";
}

// All possible interrupt types from the graph. We are using string for Reminder node
export type InterruptValue = string | number | { "question": string };

// All possible resume types to send to the graph. We are using string for Reminder node
export type ResumeValue = string | number;

File: agent/app/types/agent_types.py
from typing import TypedDict, Any, Annotated
import operator
from langgraph.graph import MessagesState

class AgentStateCtx(TypedDict):
    flow_data: str | None
    service_id: int | str | None

class Context(TypedDict):
    agent_state: AgentStateCtx
    settings: dict[str, Any]

class Weather(TypedDict):
    location: str
    search_status: str
    result: str

class State(MessagesState):
    system_prompt: str | None
    llm_core_model: str | None
    weather_forecast: Annotated[list[Weather], operator.add]
    context: Context

class WeatherInput(TypedDict):
    location: str
    tool_call_id: str

class ToolNodeArgs(TypedDict):
    name: str
    args: dict[str, Any]
    id: str

class McpToolNodeArgs(TypedDict):
    server_name: str
    name: str
    args: dict[str, Any]
    id: str